modules/localization/msf/local_pyramid_map/base_map/base_map.cc

```diff
@@ -0,0 +1,570 @@
+/******************************************************************************
+ * Copyright 2019 The Apollo Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+#include "modules/localization/msf/local_map/base_map/base_map.h"
+
+#include <set>
+#include <string>
+#include "boost/filesystem.hpp"
+#include "modules/localization/msf/common/util/file_utility.h"
+
+namespace apollo {
+namespace localization {
+namespace msf {
+
+BaseMap::BaseMap(BaseMapConfig* config)
+    : map_config_(config),
+      map_node_cache_lvl1_(nullptr),
+      map_node_cache_lvl2_(nullptr),
+      map_node_pool_(nullptr) {}
+
+BaseMap::~BaseMap() {
+  if (map_node_cache_lvl1_) {
+    delete map_node_cache_lvl1_;
+    map_node_cache_lvl1_ = nullptr;
+  }
+  if (map_node_cache_lvl2_) {
+    delete map_node_cache_lvl2_;
+    map_node_cache_lvl2_ = nullptr;
+  }
+}
+
+void BaseMap::InitMapNodeCaches(int cacheL1_size, int cahceL2_size) {
+  if (map_node_cache_lvl1_ != nullptr) {
+    delete map_node_cache_lvl1_;
+    map_node_cache_lvl1_ == nullptr;
+  }
+
+  if (map_node_cache_lvl2_ != nullptr) {
+    delete map_node_cache_lvl2_;
+    map_node_cache_lvl2_ == nullptr;
+  }
+
+  map_node_cache_lvl1_ =
```

Comment:
Consider using shared pointer to avoid manual deletion?

Reply:
fixed